### DISCUSSION TITLE
Develop endpoints/module without reload

### DISCUSSION THREAD

**Discussion by kr3niu** [directus/directus#7333] [Archive]
Hi,
It's possible to develop custom endpoints without restart directus everytime? I mean restart app with npx directus start
Should I use AMD moduke or what is recommendation?

**Comment by SimonMGarrett** [comment]
I have the same issue with developing new modules: to see my changes, I need to restart Directus. While I can totally understand that it's simpler to load things into Directus this way, it does make the development loop much longer. 

Ideally, I'd like to make a change, refresh the module page, and see the results. An alternative might be to run a CLI command to force Directus to re-load all the stuff under extensions (or the choice of just 'modules', 'endpoints', etc). That way, I could script the CLI command to run on watch and get a much shorter dev experience. 

Any chance this could be added to your roadmap? I'd guess it's going to put off a lot of developers to have a 60sec code->save->restartDirectus->viewChanges loop to develop for Directus, which would be a shame because you've built something utterly beautiful.

**Reply by nickrum** [reply]
This is already on our roadmap. Providing a CLI command to force-reload extensions would be tricky, as we would have to somehow communicate with the running API server instance. Our current plan is to provide a dedicated extension development environment that will reload API extensions and hot-module-replace App extensions when they change or a new extension has been added.

**Reply by SimonMGarrett** [reply]
Sounds like a good idea - any idea on a timeframe to deliver this? a month? three months? a year?

**Reply by nickrum** [reply]
Hard to say, as there are quite a few challenges to solve for this to work properly. Generally we do not provide any timeframes, but it will very likely be months rather than years.

**Reply by SimonMGarrett** [reply]
That's great news, many thanks Nicola.

**Reply by kr3niu** [reply]
Thanks for yours answers.
It's already any tasks our issue for that in github? 

Would be recommended workaround for now?
Should we use nodemon that restart all application with timeout?

**Reply by nickrum** [reply]
No issue for that on GitHub yet. We are still very early in the design phase of how this should work.

Your best bet is indeed to use nodemon for now. Something like `nodemon --watch extensions --signal SIGHUP --exec npx directus start` should work.

**Reply by msravi16** [reply]
Hi nickrum, 
While using "nodemon --watch extensions --signal SIGHUP --exec npx directus start"  I got error "Port 8055 is already in use".
Is it good to extend nodemon to kill 8055 port before it is restart?

**Reply by SimonMGarrett** [reply]
It should be fine - I've been using this approach for a few days, and haven't had this issue despite hundreds of restarts. Are you/have you been running Docker? If so, perhaps it's possible that's still got a version of Directus running? If not, perhaps it's another version of Directus still running in a terminal window/tab somewhere?

**Reply by msravi16** [reply]
Hi Simon,
I am running directus in windows so I extended nodemon to kill process at 8055 before restart. It is working fine.
I have not used docker.
